src/main.rs

```diff
@@ -71,23 +87,36 @@ impl RunMode {
         } else {
             let size_str = matches.opt_str("resolution").unwrap_or("1280x720".to_owned());
             let (width, height) = try!(size_str.find('x')
-                .and_then(|x| {
-                    if x == 0 || x == size_str.len() - 1 {
-                        None
-                    } else {
-                        Some((&size_str[..x], &size_str[x + 1..]))
-                    }
-                })
-                .map(|size| (size.0.parse::<u32>(), size.1.parse::<u32>()))
-                .and_then(|size| match size {
-                    (Ok(width), Ok(height)) => Some((width, height)),
-                    _ => None
-                })
-                .ok_or_else(|| GeneralError("invalid window size (WIDTHxHEIGHT)".to_owned())));
-            let fov = try!(matches.opt_str("fov").unwrap_or("65".to_owned()).parse::<f32>()
-                .map_err(|_| GeneralError("invalid value for fov".to_owned())));
-            let level = try!(matches.opt_str("level").unwrap_or("0".to_owned()).parse::<usize>()
-                .map_err(|_| GeneralError("invalid value for fov".to_owned())));
+                                               .and_then(|x| {
+                                                   if x == 0 || x == size_str.len() - 1 {
+                                                       None
+                                                   } else {
+                                                       Some((&size_str[..x], &size_str[x + 1..]))
+                                                   }
+                                               })
+                                               .map(|size| {
+                                                   (size.0.parse::<u32>(), size.1.parse::<u32>())
+                                               })
+                                               .and_then(|size| {
+                                                   match size {
+                                                       (Ok(width), Ok(height)) =>
+                                                           Some((width, height)),
+                                                       _ => None,
+                                                   }
+                                               })
+                                               .ok_or_else(|| {
+                                                   GeneralError("invalid window size \
+                                                                 (WIDTHxHEIGHT)"
+                                                                    .to_owned())
+                                               }));
```

Comment:
This chain is pretty bad, but OTOH it wasn't very good to begin with. I think I can simplify it if I have a look at it.